### PR TITLE
converter: support more options

### PR DESCRIPTION
**Pull request by imeoer**
The converter package will be imported in nydusify.

### CODE REVIEW

File: pkg/backend/oss.go [hunk 1]
```diff
@@ -63,114 +61,104 @@ func newOSSBackend(rawConfig []byte) (*OSSBackend, error) {
...
-	if chunkNum <= 0 || chunkNum > 10000 {
-		return nil, errors.New("chunkNum invalid")
+func splitFileByPartSize(blobSize, chunkSize int64) ([]oss.FileChunk, error) {
+	if chunkSize <= 0 {
+		return nil, errors.New("invalid chunk size")
 	}
 
-	if int64(chunkNum) > blobSize {
-		return nil, errors.New("oss: chunkNum invalid")
+	var chunkN = blobSize / chunkSize
```
Comment (changweige) [inline pkg/backend/oss.go:73]: It will leak the last chunk

File: pkg/backend/oss.go [hunk 2]
```diff
@@ -63,114 +61,104 @@ func newOSSBackend(rawConfig []byte) (*OSSBackend, error) {
...
+	if chunkN >= 10000 {
+		return nil, errors.New("too many parts, please increase chunk size")
```
Comment (changweige) [inline pkg/backend/oss.go:75]: Is there any principle why 10000 is chosen as the threshold?

Comment (changweige) [inline pkg/backend/oss.go:75]: How user can increase chunks size, it seems hard-coded as 500MB?

File: pkg/backend/oss.go [hunk 3]
```diff
@@ -63,114 +61,104 @@ func newOSSBackend(rawConfig []byte) (*OSSBackend, error) {
...
 	}
 
 	var chunks []oss.FileChunk
 	var chunk = oss.FileChunk{}
-	var chunkN = (int64)(chunkNum)
 	for i := int64(0); i < chunkN; i++ {
 		chunk.Number = int(i + 1)
```
Comment (changweige) [inline pkg/backend/oss.go:81]: Why is there a offset `1` between chunk number and the index?

File: pkg/converter/types.go [hunk 4]
```diff
@@ -66,6 +66,10 @@ type PackOption struct {
 	Compressor string
 	// OCIRef enables converting OCI tar(.gz) blob to nydus referenced blob.
 	OCIRef bool
+	// AlignedChunk aligns uncompressed data chunks to 4K, only for RAFS V5.
```
Comment (changweige) [inline pkg/converter/types.go:69]: Why do we need chunks to be aligned by 4K for rafs v5?

File: pkg/backend/oss.go (hunk 1 above)
Comment (imeoer) [reply pkg/backend/oss.go:73]: It will be appended later.

File: pkg/backend/oss.go (hunk 2 above)
Comment (imeoer) [reply pkg/backend/oss.go:75]: The code is copied from aliyun-oss-go-sdk, I will add a port comment.

File: pkg/backend/oss.go (hunk 3 above)
Comment (imeoer) [reply pkg/backend/oss.go:81]: It's defined by `oss.FileChunk`.

File: pkg/converter/types.go (hunk 4 above)
Comment (imeoer) [reply pkg/converter/types.go:69]: It's compatible for `nydusify convert --aligned-chunk`.

File: pkg/backend/oss.go (hunk 1 above)
Comment (changweige) [reply pkg/backend/oss.go:73]: Can we have a comment telling the trick at least?

File: pkg/converter/types.go (hunk 4 above)
Comment (changweige) [reply pkg/converter/types.go:69]: I think we don't have to expose the parameter `aligned-chunk`. Rafs v5 does not have to be aligned and rafs v6 must be aligned. Perhaps, the parameter can be removed both from nydusify and nydus-image

File: pkg/backend/oss.go (hunk 2 above)
Comment (imeoer) [reply pkg/backend/oss.go:76]: Updated, users can set `backend.MultipartChunkSize = X` to change the chunk size.

File: pkg/backend/oss.go (hunk 1 above)
Comment (imeoer) [reply pkg/backend/oss.go:74]: It's ported from https://github.com/aliyun/aliyun-oss-go-sdk/blob/v2.2.6/oss/utils.go#L259, leave the port code as unchanged as possible.

File: pkg/converter/types.go (hunk 4 above)
Comment (imeoer) [reply pkg/converter/types.go:69]: We still need to keep the compatibility, at least it doesn't have any broken changes for minor release version, right?

Comment (changweige) [reply pkg/converter/types.go:69]: Um. there is no compatibility problem no matter whether aligned-chunk is exposed or not.

Comment (imeoer) [reply pkg/converter/types.go:69]: The converter package will be used in nydusify v2.2.

**Review by changweige — APPROVED** [review]
LGTM

**Pull request merged**